Drop-In/src/main/java/com/braintreepayments/api/dropin/VaultManagerActivity.java

```diff
@@ -0,0 +1,143 @@
+        setResult(Activity.RESULT_OK);
+    }
+
+    @Override
+    public void onError(Exception error) {
+        if(error instanceof PaymentMethodDeleteException) {
+            PaymentMethodDeleteException exception = (PaymentMethodDeleteException)error;
+            PaymentMethodNonce paymentMethodNonce = exception.getPaymentMethodNonce();
+
+            mAdapter.cancelSwipeOnPaymentMethodNonce(paymentMethodNonce);
+
+            Snackbar.make(findViewById(R.id.bt_base_view), R.string.bt_vault_manager_delete_failure,
+                    Snackbar.LENGTH_LONG).show();
+            mBraintreeFragment.sendAnalyticsEvent("manager.delete.failed");
+        } else {
+            mBraintreeFragment.sendAnalyticsEvent("manager.unknown.failed");
+            finish(error);
+        }
+    }
+
+    @Override
+    public void onClick(View v) {
+        if (v instanceof PaymentMethodItemView) {
+            final AtomicBoolean positiveSelected = new AtomicBoolean(false);
+
+            PaymentMethodItemView paymentMethodItemView = (PaymentMethodItemView)v;
+            final PaymentMethodNonce paymentMethodNonceToDelete = paymentMethodItemView
+                    .getPaymentMethodNonce();
+
+            PaymentMethodItemView dialogView = new PaymentMethodItemView(this);
+            dialogView.setPaymentMethod(paymentMethodNonceToDelete, false);
+
+            new AlertDialog.Builder(VaultManagerActivity.this,
+                    R.style.Theme_AppCompat_Light_Dialog_Alert)
+                    .setTitle(R.string.bt_delete_confirmation_title)
+                    .setMessage(R.string.bt_delete_confirmation_description)
+                    .setView(dialogView)
+                    .setPositiveButton(R.string.bt_delete, new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            positiveSelected.set(true);
+                            mBraintreeFragment.sendAnalyticsEvent("manager.delete.confirmation.positive");
+                            PaymentMethod.deletePaymentMethod(mBraintreeFragment, paymentMethodNonceToDelete);
+                        }
+                    })
+                    .setOnDismissListener(new DialogInterface.OnDismissListener() {
+                        @Override
+                        public void onDismiss(DialogInterface dialog) {
+                            if (!positiveSelected.get()) {
+                                mAdapter.cancelSwipeOnPaymentMethodNonce(paymentMethodNonceToDelete);
+                                mBraintreeFragment.sendAnalyticsEvent("manager.delete.confirmation.negative");
+                            }
+                        }
+                    })
+                    .setNegativeButton(R.string.bt_cancel, new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
```

Comment:
Will this dismiss the dialog?

Reply:
Yes. `AlertDialog.Builder` will call the `OnDismissListener` on either button click. So within that listener we check if the positive button wasn't selected and cancel the swipe.

Reply:
🙏  thanks Yash!